platform/broadcom/docker-syncd-brcm/start.sh

```diff
@@ -9,10 +9,20 @@ rm -f ${SYNCD_SOCKET_FILE}
 
 mkdir -p /etc/sai.d/
 
+# There are two ways to specify the contents of the SAI_INIT_CONFIG_FILE and they are mutually exclusive
+# via current method (sai.profile.j2) or new method (config.bcm.j2)
+# If delta is large, use sai.profile.j2 which basically require the user to select which config file to use
+# If delta is small, use config.bcm.j2 where additional SAI INIT config properties are added
+#   based on specific device metadata requirement
+#   in this case sai.profile should have been modified to use the path /etc/sai.d/config.bcm
+
 # Create/Copy the sai.profile to /etc/sai.d/sai.profile
 if [ -f $HWSKU_DIR/sai.profile.j2 ]; then
     sonic-cfggen -d -t $HWSKU_DIR/sai.profile.j2 > /etc/sai.d/sai.profile
 else
+    if [ -f $HWSKU_DIR/config.bcm.j2 ]; then
+        sonic-cfggen -d -t $HWSKU_DIR/config.bcm.j2 > /etc/sai.d/config.bcm
+    fi
```

Comment:
We could re-organize this whole if structure to:

if has config.bcm.j2
   generate config.bcm and sai.profile.
elif has sai.profile.j2
   generate sai.profile
elif has sai.profile
   copy it.
endif

Reply:
@yxieca 
We also have to handle the case where there are no config.bcm.j2 nor sai.profile.j2.
in that case we just need to copy the sai.profile from HWSKU_DIR to the new /etc/sai.d directory.

So that is why the current start.sh changes is changed to also cover that case.
When I mentioned config.bcm.j2 and sai.profile.j2 are mutual exclusive, I should have mentioned that there is also a third case where both are absent that needs to be handled. I will update the description to address that.
Let me know if you are ok with this and if you have other concerns...

Reply:
The 3rd case is handled by a static sai.profile in the device folder, right? I think that if the config.bcm is auto generated, and the file name is fixed, there is no reason to leave a chance for sai.profile to be inconsistent.

If you need to handle case of generating both bcm.config and sai.profile, then you can use
if has config.bcm.j2:
   generate config.bcm and sai.profile
endif
if has sai.profile.j2
  generate sai.profile
endif
# 3rd case: use existing sai.profile in device folder.

Reply:
We should not use both config.bcm.j2 and sai.profile.j2.  It unnecessarily complicate things and error pron.
That is why I mentioned user should pick one or the other(mutually exclusive).
In fact, eventually we could retire the usage of sai.profile.j2 if the difference between the config files is small. and move on to use config.bcm.j2 to eliminate having multiple large config files with very small differences.  this is why Guohan asked me to make this new change.  My initial PR was using sai.profile.j2 as well and as you can see if I used that approach we will now end up with 3 large config files for "Arista-7260CX3-C64" with very small delta between them.  Two of them differ from the MMU (T0/T1) and another one for the IPinIP SOC properties if DualToR...